### PR TITLE
[Fixes #6514] Add default ctor overload to AuthorizeFilter

### LINKED ISSUE #6514: Add default ctor overload to AuthorizeFilter
Currently creating AuthorizeFilter requires specifying a policy. We should provide a default overload that is equivalent to specifying `[Authorize]` everywhere.

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: test/WebSites/SecurityWebSite/Program.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static void Main(string[] args)
             var host = new WebHostBuilder()
                 .UseKestrel()
                 .UseContentRoot(Directory.GetCurrentDirectory())
-                .UseStartup<Startup>()
+                .UseStartup<StartupWithGlobalDenyAnonymousFilter>()
```
Comment (kichalla) [inline test/WebSites/SecurityWebSite/Program.cs]: Oops, changed it my mistake. Will revert this.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Authorization/AuthorizeFilterTest.cs [hunk 2]
```diff
@@ -26,6 +26,21 @@ public void InvalidUser()
             Assert.Contains(authorizationContext.HttpContext.User.Identities, i => i.IsAuthenticated);
         }
 
+        [Fact]
+        public async Task AuthorizeFilter_CreatedWithDefaultConstructor_ThrowsWhenOnAuthorizationAsyncIsCalled()
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Authorization/AuthorizeFilterTest.cs]: Can we make a more meaningful test than this? It's not clear why this is an error case.

File: test/WebSites/SecurityWebSite/StartupWithGlobalDenyAnonymousFilter.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                .AddCookie(options =>
+                {
+                    options.LoginPath = "/Home/Login";
+                    options.LogoutPath = "/Home/Logout";
+                });
+
+            services.AddMvc(o =>
+            {
+                o.Filters.Add(new AuthorizeFilter());
+            });
```
Comment (rynowak) [inline test/WebSites/SecurityWebSite/StartupWithGlobalDenyAnonymousFilter.cs]: Cool beans

**Comment by kichalla** [comment]
🆙 📅

**Pull request merged**